src/core.ts

```diff
@@ -243,14 +243,15 @@ function uploadPendingEvents() {
 }
 
 function init() {
-  // Set ClarityID cookie, if it's not set already
-  if (!getCookie(Cookie)) {
-    // setting our ClarityId cookie for 2 years
-    setCookie(Cookie, guid(), 7 * 52 * 2);
-  }
-  cid = getCookie(Cookie);
+  let cidCookie = getCookie(Cookie);
+  cid = cidCookie || guid();
   impressionId = guid();
 
+  // Set 2-year CID cookie, if allowed by config
+  if (!cidCookie && config.allowIdCookie) {
+    setCookie(Cookie, cid, 7 * 52 * 2);
```

Comment:
>7 * 52 * 2 [](start = 27, length = 10)

Minor but make this a configuration and just use a null or zero value to indicate no cookies.